File: qc-core/src/main/java/com/eager/questioncloud/core/domain/question/QuestionFilter.java
package com.eager.questioncloud.core.domain.question;

import com.eager.questioncloud.core.common.PagingInformation;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class QuestionFilter {
    private Long userId;
    private List<Long> categories;
    private List<QuestionLevel> levels;
    private QuestionType questionType;
    private Long creatorId;
    private QuestionSortType sort;
    private PagingInformation pagingInformation;
}

File: qc-core/qc-question/qc-question-core/src/main/kotlin/com/eager/questioncloud/question/common/QuestionFilter.kt
package com.eager.questioncloud.question.common

import com.eager.questioncloud.question.enums.QuestionLevel
import com.eager.questioncloud.question.enums.QuestionSortType
import com.eager.questioncloud.question.enums.QuestionType

class QuestionFilter(
    val categories: List<Long>? = null,
    val levels: List<QuestionLevel>? = null,
    val questionType: QuestionType? = null,
    val creatorId: Long? = null,
    val sort: QuestionSortType,
)
